vcr/trust/trust.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package trust
+
+import (
+	"errors"
+	"os"
+	"sync"
+
+	"github.com/nuts-foundation/go-did"
+	"gopkg.in/yaml.v2"
+)
+
+// Config holds the trusted issuers per credential type
+type Config struct {
+	filename       string
+	issuersPerType map[string][]string
+	mutex          sync.Mutex
+}
+
+// NewConfig returns a fully configured Config
+func NewConfig(filename string) Config {
+	return Config{
+		filename:       filename,
+		issuersPerType: map[string][]string{},
+		mutex:          sync.Mutex{},
+	}
+}
+
+// Load the trusted issuers per credential type from file
+func (tc Config) Load() error {
```

Comment:
now the mutex is part of the struct you pass it by value when calling `Load`, `AddTrust` or `RemoveTrust` which IntelliJ warns about:

`'Load' passes lock by value: type 'Config' contains 'sync.Mutex' which is 'sync.Locker' `

You have to make define it as pointer in the struct or use the pointer receiver on the funcs

Reply:
I'll pay closer attention to the tiny picto in the upper right...

Reply:
:-p